Common/Field/src/MagFieldFast.cxx

```diff
@@ -124,12 +124,40 @@ bool MagFieldFast::Field(const double xyz[3], double bxyz[3]) const
 bool MagFieldFast::GetBcomp(EDim comp, const double xyz[3], double& b) const
 {
   // get field
-  const float fxyz[3] = { float(xyz[0]), float(xyz[1]), float(xyz[2]) };
   int zSeg, rSeg, quadrant;
-  if (!GetSegment(fxyz, zSeg, rSeg, quadrant))
+  if (!GetSegment(xyz[kX],xyz[kY],xyz[kZ], zSeg, rSeg, quadrant)) {
     return false;
+  }
   const SolParam* par = &mSolPar[rSeg][zSeg][quadrant];
-  b = CalcPol(par->parBxyz[comp], fxyz[kX], fxyz[kY], fxyz[kZ]) * mFactorSol;
+  b = CalcPol(par->parBxyz[comp], xyz[kX], xyz[kY], xyz[kZ]) * mFactorSol;
+  //
+  return true;
+}
+
+//_______________________________________________________________________
+bool MagFieldFast::GetBcomp(EDim comp, const Point3D<float> xyz, double& b) const
```

Comment:
not 100% sure ... but it might be slightly faster to pass a const reference to Point3D<float> (unless this is an inline function)

Reply:
I've done some testing on reasonably recent hardware and compilers - there is no benefit up to about 16 bytes, so with ~3 floats it should be fine by value. Another benefit of having a by-value interface is copy elision (guaranteed in C++17) which would kick in if you e.g. pass in a temporary like an in place {}-initialized Point3D.